Mathlib/Analysis/SpecialFunctions/Gaussian.lean

```diff
@@ -47,21 +47,67 @@ open scoped Real Topology FourierTransform
 
 open Complex hiding exp continuous_exp abs_of_nonneg sq_abs
 
+theorem exp_neg_mul_rpow_isLittleO_exp_neg {p b : ℝ} (hb : 0 < b) (hp : 1 < p) :
+    (fun x : ℝ => exp (- b * x ^ p)) =o[atTop] fun x : ℝ => exp (-x) := by
+  rw [isLittleO_exp_comp_exp_comp]
+  suffices Tendsto (fun x => x * (b * x ^ (p - 1) + -1)) atTop atTop by
+    refine Tendsto.congr' ?_ this
+    refine eventuallyEq_of_mem (Ioi_mem_atTop (0 : ℝ)) (fun x hx => ?_)
+    rw [rpow_sub_one (ne_of_gt hx)]
+    field_simp [(by exact ne_of_gt hx : x ≠ 0)]
+    ring
+  apply Tendsto.atTop_mul_atTop tendsto_id
+  refine tendsto_atTop_add_const_right atTop (-1 : ℝ) ?_
+  exact Tendsto.const_mul_atTop hb (tendsto_rpow_atTop (by linarith))
+
 theorem exp_neg_mul_sq_isLittleO_exp_neg {b : ℝ} (hb : 0 < b) :
     (fun x : ℝ => exp (-b * x ^ 2)) =o[atTop] fun x : ℝ => exp (-x) := by
-  have A : (fun x : ℝ => -x - -b * x ^ 2) = fun x => x * (b * x + -1) := by ext x; ring
-  rw [isLittleO_exp_comp_exp_comp, A]
-  apply Tendsto.atTop_mul_atTop tendsto_id
-  exact tendsto_atTop_add_const_right atTop (-1 : ℝ) (Tendsto.const_mul_atTop hb tendsto_id)
+  simp_rw [← rpow_two]
+  exact exp_neg_mul_rpow_isLittleO_exp_neg hb one_lt_two
 #align exp_neg_mul_sq_is_o_exp_neg exp_neg_mul_sq_isLittleO_exp_neg
 
+theorem rpow_mul_exp_neg_mul_rpow_isLittleO_exp_neg (s : ℝ) {b p : ℝ} (hp : 1 < p) (hb : 0 < b) :
+    (fun x : ℝ => x ^ s * exp (- b * x ^ p)) =o[atTop] fun x : ℝ => exp (-(1 / 2) * x) := by
+  apply ((isBigO_refl (fun x : ℝ => x ^ s) atTop).mul_isLittleO
+      (exp_neg_mul_rpow_isLittleO_exp_neg hb hp)).trans
+  simpa only [mul_comm] using Real.Gamma_integrand_isLittleO s
+
 theorem rpow_mul_exp_neg_mul_sq_isLittleO_exp_neg {b : ℝ} (hb : 0 < b) (s : ℝ) :
     (fun x : ℝ => x ^ s * exp (-b * x ^ 2)) =o[atTop] fun x : ℝ => exp (-(1 / 2) * x) := by
-  apply ((isBigO_refl (fun x : ℝ => x ^ s) atTop).mul_isLittleO
-      (exp_neg_mul_sq_isLittleO_exp_neg hb)).trans
-  simpa only [mul_comm] using Gamma_integrand_isLittleO s
+  simp_rw [← rpow_two]
+  exact rpow_mul_exp_neg_mul_rpow_isLittleO_exp_neg s one_lt_two hb
 #align rpow_mul_exp_neg_mul_sq_is_o_exp_neg rpow_mul_exp_neg_mul_sq_isLittleO_exp_neg
 
+-- TODO. do integrableOn_rpow_mul_exp_neg_mul_rpow
```

Comment:
Could you do the todo? It shouldn't be much harder, given that you already have the previous lemmas.

Reply:
I did it using `integrableOn_rpow_mul_exp_neg_rpow` since otherwise I could not get the case `p = 1` using the lemma `rpow_mul_exp_neg_mul_rpow_isLittleO_exp_neg`. 

Note that this is not my domain of expertise and it is quite possible (probable?) that some proofs are not optimal.